x-pack/plugins/cross_cluster_replication/server/types.ts

```diff
@@ -25,24 +24,9 @@ export interface StartDependencies {
 }
 
 export interface RouteDependencies {
-  router: CcrPluginRouter;
+  router: IRouter<RequestHandlerContext>;
```

Comment:
nit: `RequestHandlerContext` is the default value of the generics in the `IRouter` interface
```suggestion
  router: IRouter;
```